server/commands/write.go

```diff
@@ -60,9 +60,37 @@ func (c *WriteCommand) validateTuplesets(ctx context.Context, req *openfgapb.Wri
 	}
 
 	for _, tk := range writes {
-		if _, err := tupleUtils.ValidateTuple(ctx, c.datastore, store, modelID, tk, dbCallsCounter); err != nil {
+		tupleUserset, err := tupleUtils.ValidateTuple(ctx, c.datastore, store, modelID, tk, dbCallsCounter)
+		if err != nil {
 			return serverErrors.HandleTupleValidateError(err)
 		}
+		switch usType := tupleUserset.Userset.(type) {
+		case *openfgapb.Userset_This:
+			continue // no need to check on Direct Relationship
+		case *openfgapb.Userset_Intersection:
+			isDirect := isDirectIntersection(usType, tk)
```

Comment:
If you would like you could now remove the extra variable `isDirect` and just write it as
```
if !isDirectIntersection(usType, tk) { ... }
```

Reply:
Great point -- simplifies it some more, and is in the latest commit 👍